warehouse/query-core/src/main/java/datawave/query/tracking/ActiveQueryLog.java

```diff
@@ -4,83 +4,139 @@
 import java.util.UUID;
+import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ActiveQueryLog {
     
+    public static final String DEFAULT_NAME = "default_log";
     private static final String DEFAULT_EMPTY_QUERY_ID = new UUID(0, 0).toString();
-    private static Logger log = Logger.getLogger(ActiveQueryLog.class);
-    private static ActiveQueryLog instance = null;
+    private static final Logger log = Logger.getLogger(ActiveQueryLog.class);
+    private static Cache<String,ActiveQueryLog> logCache = null;
+    private static final Object logCacheLock = new Object();
     private static AccumuloConfiguration conf = null;
     
     // Accumulo properties
     public static final String MAX_IDLE = "datawave.query.active.maxIdleMs";
     public static final String LOG_PERIOD = "datawave.query.active.logPeriodMs";
     public static final String LOG_MAX_QUERIES = "datawave.query.active.logMaxQueries";
     public static final String WINDOW_SIZE = "datawave.query.active.windowSize";
-    private static long HOURS_24_MS = TimeUnit.MILLISECONDS.convert(24, TimeUnit.HOURS);
-    private static long MINUTES_15_MS = TimeUnit.MILLISECONDS.convert(15, TimeUnit.MINUTES);
-    private static long MINUTES_1_MS = TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
+    private static final long HOURS_24_MS = TimeUnit.MILLISECONDS.convert(24, TimeUnit.HOURS);
+    private static final long MINUTES_15_MS = TimeUnit.MILLISECONDS.convert(15, TimeUnit.MINUTES);
+    private static final long MINUTES_1_MS = TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES);
     
     // Changeable via Accumulo properties
     volatile private long maxIdle = MINUTES_15_MS;
     volatile private long logPeriod = MINUTES_1_MS;
     volatile private int logMaxQueries = 5;
     volatile private int windowSize = 10;
-    private AtomicLong lastAccess = new AtomicLong(System.currentTimeMillis());
+    private final AtomicLong lastAccess = new AtomicLong(System.currentTimeMillis());
     
     private Cache<String,ActiveQuery> CACHE = null;
-    private ReentrantReadWriteLock cacheLock = new ReentrantReadWriteLock();
+    private final ReentrantReadWriteLock cacheLock = new ReentrantReadWriteLock();
     private Timer timer = null;
     private TimerTask timerTask = null;
     
-    synchronized public static void setConfig(AccumuloConfiguration conf) {
+    private final String name;
+    
+    synchronized public static void setConfig(final AccumuloConfiguration conf) {
         if (conf != null) {
             if (ActiveQueryLog.conf == null || conf.getUpdateCount() > ActiveQueryLog.conf.getUpdateCount()) {
                 ActiveQueryLog.conf = conf;
-                ActiveQueryLog.getInstance().checkSettings(conf, false);
+            }
+            // Do not allow access to the cache while updating each log's settings.
```

Comment:
This is where I thought the  if (logCache != null) check could be outside of the synchronized

Reply:
The other changes look fine.  I just have these two issues with the synchronization changes in AQL.

Reply:
Fixed.